x/staging/posts/keeper/grpc_query.go

```diff
@@ -153,9 +153,30 @@ func (k Keeper) RegisteredReactions(goCtx context.Context, _ *types.QueryRegiste
 func (k Keeper) Reports(
 	ctx context.Context, request *types.QueryReportsRequest,
 ) (*types.QueryReportsResponse, error) {
+	var reports []types.Report
 	sdkCtx := sdk.UnwrapSDKContext(ctx)
-	reports := k.GetPostReports(sdkCtx, request.PostId)
-	return &types.QueryReportsResponse{Reports: reports}, nil
+
+	store := sdkCtx.KVStore(k.storeKey)
+	reportsStore := prefix.NewStore(store, types.ReportsStorePrefix)
```

Comment:
I'm not sure this will work. I think it would be much better to simply use `prefix.NewStore(store, types.ReportStoreKey(postID))` and then remove the below check of whether the report has post id equals to the given one.

 Also, we should add some tests to make sure they are returned properly